disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/exporters/AbstractDiskExporter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.internal.exporters;
+
+import io.opentelemetry.contrib.disk.buffering.internal.serialization.serializers.SignalSerializer;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.FolderManager;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.Storage;
+import io.opentelemetry.contrib.disk.buffering.storage.StorageConfiguration;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public abstract class AbstractDiskExporter<EXPORT_DATA> {
+  private final Storage storage;
+  private static final Logger logger = Logger.getLogger(AbstractDiskExporter.class.getName());
+
+  public AbstractDiskExporter(File rootDir, StorageConfiguration configuration) {
+    validateConfiguration(configuration);
+    this.storage = new Storage(new FolderManager(getSignalFolder(rootDir), configuration));
+  }
+
+  /**
+   * Reads data from the disk and attempts to export it.
+   *
+   * @param timeout The amount of time to wait for the wrapped exporter to finish.
+   * @param unit The unit of the time provided.
+   * @return TRUE if there was data available and it was successfully exported within the timeout
+   *     provided. FALSE if either of those conditions didn't meet.
+   * @throws IOException If an unexpected error happens.
+   */
+  public boolean exportStoredBatch(long timeout, TimeUnit unit) throws IOException {
+    logger.log(Level.INFO, "Attempting to export batch from disk.");
+    AtomicBoolean exportSucceeded = new AtomicBoolean(false);
+    boolean foundDataToExport =
+        storage.read(
+            bytes -> {
+              logger.log(Level.INFO, "About to export stored batch.");
+              CompletableResultCode join =
+                  doExport(getSerializer().deserialize(bytes)).join(timeout, unit);
+              exportSucceeded.set(join.isSuccess());
```

Comment:
what's the reason for the atomic boolean?

Reply:
Since I needed to know the outcome of the delegated exporter by line 51, the IDE suggested either using AtomicBoolean or an array of booleans with a single item to be able to change its value in a callback and read it outside of the callback later.

Reply:
can we simply to?

we'd have to pass the result of the consumer back to the caller in https://github.com/LikeTheSalad/opentelemetry-java-contrib/blob/9ab9aa91e7a92e0e4e208c794da16bc9e4cdf1e5/disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/storage/files/ReadableFile.java#L81 - but I think that would be easier to understand

```java
    return storage.read(
        bytes -> {
          logger.log(Level.INFO, "About to export stored batch.");
          CompletableResultCode join =
              doExport(getSerializer().deserialize(bytes)).join(timeout, unit);
          return join.isSuccess();
        });
```

Reply:
I see, that looks way more readable for sure. I'll apply the changes, though maybe I'll also rename the `read` method to something like `readAndProcess` in order to express the full intention and probably give a hint on what to expect as return from it.

Reply:
Thank you, I've made the changes.